shipyard_blueprints/googledrive/shipyard_googledrive/drive_utils.py

```diff
@@ -269,15 +287,146 @@ def get_file_matches(
     Returns: The list of the matches
 
     """
-    folder_id = get_folder_id(folder_id, service=service)
     try:
-        query = f"name contains '{pattern}'"
+        # query = f"name contains '{pattern}'"
+        query = None
         if folder_id:
-            query += f" and '{folder_id}' in parents"
-        results = service.files().list(q=query).execute()
-        files = results.get("files", [])
+            # query += f" and '{folder_id}' in parents"
```

Comment:
I am going to butcher explaining this but you can clean up the code by setting the query based on if the folder id is provided. that should make one of the else blocks redundant. I would also test if passing None into the functions yeilds the same results. for example if
```.list(
                            q=query,
                            supportsAllDrives=True,
                            includeItemsFromAllDrives=True,
                            corpora="drive",
                            driveId=drive_id,
                        )
```
the same as            
```.list(
                            q=query,
                            supportsAllDrives=True,
                            includeItemsFromAllDrives=True,
                            corpora="drive",
                            driveId=None,
                        )
```

if that is the case you could clean up this code

Reply:
The issue here is that the other parameters like supportsAllDrives, corpora, includeItemsFromAllDrives are only necessary if the `drive_id` is provided, so that's why I had to handle all of those separately